internal/common/shlex_quote.go

```diff
@@ -0,0 +1,13 @@
+package common
+
+import (
+	"strings"
+)
+
+// ShlexQuote implements a variant of pythons shlex.quote()
+// in go (c.f. https://github.com/python/cpython/blob/3.14/Lib/shlex.py#L320)
```

Comment:
Let's be clear here what the variation is; we're not checking the table of 'safe characters' and that's the only deviation right?

Reply:
I can expand this, the idea is really to just drop the optimization in the python code to skip qutoting if only "safe" chars are used. We could do that too but I don't (currently?) see why.